### ISSUE TITLE
Change flow of combining of multiple raw data files

### ISSUE THREAD

**Issue by leewujung** [OSOceanAcoustics/echopype#181]
Currently when multiple raw data files are fed into `Convert`, the files are converted sequentially. This process in theory can be parallelized without much problem. 

When `combine_opt=True`, the current implementation for `raw2nc` converts each file into a temporary files and combine all files at the end by `xarray.open_mfdataset`, while `raw2zarr` reads the files sequentially and append the new data section into the combined file. The difference was due to xarray not supporting appending data slices to existing netCDF files. 

Therefore, to enable parallel conversion, we can just change how each file is handled under the hood for `raw2nc`. This is something we can do right now.

Things are a bit more complicated for `raw2zarr`, because zarr engine for `open_mfdataset` is currently [a PR under dev](https://github.com/pydata/xarray/pull/4187).  Let's make sure to come back to this once this PR is merged.

**Comment by leewujung** [comment]
Updates:
- We have made this under the hood change on how multiple raw files are converted and merged. The current structure makes it straightforward to enable parallel conversion. I am changing what this issue entails and move the parallel conversion part to another issue.
- The zarr engine for open_mfdataset has been added and we have updated echopype to use the new functionality. Now conversion to netCDF and zarr are interchangeable.

This issue can be closed once we have class-redesign merged to master.
